cclib/parser/turbomoleparser.py

```diff
@@ -208,6 +227,51 @@ def extract(self, inputfile, line):
             self.set_attribute('vibirs', vibirs)
             self.set_attribute('vibdisps', vibdisps)
 
+        # In this section we are parsing mocoeffs and moenergies from
+        # the files like: mos, alpha and beta.
+        # $scfmo    scfconv=6   format(4d20.14)
+        # # SCF total energy is     -382.3457535740 a.u.
+        # #
+        #      1  a      eigenvalue=-.97461484059799D+01   nsaos=60
+        # 0.69876828353937D+000.32405121159405D-010.87670894913921D-03-.85232349313288D-07
+        # 0.19361534257922D-04-.23841194890166D-01-.81711001390807D-020.13626356942047D-02
+        # ...
+        # ...
+        # $end
+        if (line.startswith('$scfmo') or line.startswith('$uhfmo')) and line.find('scfconv') > 0:
+            if line.strip().startswith('$uhfmo_alpha'):
+                self.unrestricted = True
+
+            while not (line.strip().startswith('1') and line.find('eigenvalue') > 0):
+                line = next(inputfile)
+
+            moenergies = []
+            mocoeffs = []
+
+            while not line.strip().startswith('$'):
+                info = line.split()
+                eigenvalue = float(info[2][11:].replace('D', 'E'))

```

Comment:
You can use `self.float`: https://github.com/cclib/cclib/blob/180f02ccc556b6e96f5f0e9b4593ddeb7c383dcd/cclib/parser/logfileparser.py#L393

Reply:
How is it different from just using `float()`?

Reply:
If you look at the code, `self.float` does basically everything you're doing here.